### ISSUE TITLE
Allow for finer control over legend position/type

### ISSUE THREAD

**Issue by ClementTsang** [ClementTsang/bottom#1125]
### Checklist

- [X] I've looked through [the documentation](https://clementtsang.github.io/bottom/nightly/) and  [existing open issues](https://github.com/ClementTsang/bottom/issues?q=is%3Aopen+is%3Aissue+label%3Afeature) for this feature/feature request.

### Describe the feature request

For all time graphs, it would be nice to be able to have finer-grain control over how the legend is displayed (e.g. which side it appears, the type, like whether it's like the CPU one or normal ones, hiding it, etc).

**Comment by RobertoMaurizzi** [comment]
It would be useful to have the memory and network charts legends on the left too, maybe even by default, since right now it often happen you don't see anything until a few minutes after it happens because the chart lines are hidden by the legend.

**Comment by JustCauseWhyNot** [comment]
I'll add my request here since it seems more appropriate here than in a separate issue. I'd be nice to have an option to choose to show either the box of info in mem/net, over using the graph. Especially so if using a tiling wm such as I, and you've got a lot of windows open. That way when the window for btm gets shrunk I'd see the number like this
![2023-11-23_12-48](https://github.com/ClementTsang/bottom/assets/58643844/3480d134-b4ca-4086-98bd-5ac221140287) rather than just seeing a graph which isn't all the useful when in small window.

**Comment by lyuha** [comment]
I completely agree this comment. So I make available that the chart can have a left legend at ratatui v0.25(https://github.com/ratatui-org/ratatui/pull/378). 

> It would be useful to have the memory and network charts legends on the left too, maybe even by default, since right now it often happen you don't see anything until a few minutes after it happens because the chart lines are hidden by the legend.

**Comment by lyuha** [comment]
I tested that chart draw a legend on left side. How about it?

![image](https://github.com/ClementTsang/bottom/assets/4014016/b45e9f03-81f5-4e24-aea5-38bedadf832b)
![image](https://github.com/ClementTsang/bottom/assets/4014016/4ea17f17-0a65-42ff-8b90-8bbdd216ad3e)

**Comment by RobertoMaurizzi** [comment]
I'd say it's definitely more useful, thanks @lyuha!

**Comment by lyuha** [comment]
I read the refactoring commit(77777ef5887c2e1bd1a10c55cbb329c0471d4388).
Now the `TimeChart` has an option to configure it's legend position. However, Bottom doesn't have any config option about a legend position.

Is there any plan about this?
Add options like cpu's `left_side` in config file, add key layout section in config? or have any other?

**Comment by ClementTsang** [comment]
Yep, was planning on adding this at some point.

**Comment by ClementTsang** [comment]
If you want you can try to make a PR, but it's mostly figuring out how the config/arg option would look really.
